### ISSUE TITLE
Autocomplete for variable duplicates its name

### ISSUE THREAD

**Issue by polo7** [zed-industries/zed#12830]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When you start typing a name of variable and use autocomplete then the result will consist of "typed-part" + "full name added by autocomplete".

Example: 
var name
I type "na" and get [name] in the hint -> choose this offer and get: **na**name

```
func main() {
	var sum int
	var difference int
	difference (<- well autocompleted), ssum (<- autocompleted with double "s")
}
```

This time the bug appears only for 2nd, 3d and so on variables in a row. 

https://github.com/zed-industries/zed/assets/40734771/5e4766c1-30b3-47fd-82d5-3fb31d525eb4

But sometimes it appears even with the first variable in a row (I guess there is something with parsing of unfinished lines)

https://github.com/zed-industries/zed/assets/40734771/0762efa6-4d6a-4bb6-a3ce-8cd5bd758680

### Environment

Zed: v0.138.6 (Zed)
OS: macOS 12.7.3
Memory: 8 GiB
Architecture: x86_64

**Comment by osiewicz** [comment]
Hey, thanks for the report. I believe this is not a bug (kinda); gopls doesn't support insert and replace edits it seems.
See #9634 and #10411.
TL;DR: yes, that behavior is wrong, but it can be solved within the language server itself.

**Comment by osiewicz** [comment]
Whoops, I take that back. It seems to support it, based on the sources alone: https://github.com/golang/tools/blob/03419175878954be2b792c470f49819e7e0f616e/gopls/internal/protocol/tsinsertreplaceedit.go#L25

**Comment by osiewicz** [comment]
And there's your answer; they do have support for it in source, but never return it in fact: https://github.com/search?q=repo%3Agolang%2Ftools%20SelectCompletionTextEdit&type=code (see how the `SelectCompletionTextEdit` func is never called with `true` parameter for `useReplaceMode`)

**Comment by osiewicz** [comment]
Ok, yeah, sorry, I didn't read the issue thoroughly enough. gopls does support replacement text edits, but it is funky at times, as you've spotted. The same issue is reproducible with VScode as well. I'd say this is not an issue with Zed, but with the language server itself. 

Let's keep this issue open just in case anything changes on gopls side.

**Comment by osiewicz** [comment]
Potentially related: https://go-review.googlesource.com/c/tools/+/585275

**Comment by someone13574** [comment]
I've noticed this in Rust as well.

**Comment by polo7** [comment]
> Hi there! 👋 We're working to clean up our issue tracker by closing older issues that might not be relevant anymore. If you are able to reproduce this issue in the latest version of Zed, please let us know by commenting on this issue, and we will keep it open. If you can't reproduce it, feel free to close the issue yourself. Otherwise, we'll close it in 7 days. Thanks for your help!

Yes, it is still there. See screenshot attached.

![Image](https://github.com/user-attachments/assets/482d5509-296f-47ee-b480-f68dc18686fb)

**Comment by maxhoffmann** [comment]
yep, same issue here on lastest Zed and macOS. always ending up with `$$var` when autocompleting a variable. always need to go back and delete one

**Comment by facelessfish** [comment]
Yes, still there. It's the main issue stopping me from actually using Zed.

**Comment by patzm** [comment]
still relevant 😭
